test/e2e/devices/devicecreatetest/device_create_test.go

```diff
@@ -42,46 +42,65 @@ func TestCli_Devices_Create(t *testing.T) {
 			want: &cobra.Command{},
 			cmdFunc: func(t *testing.T, c *cobra.Command) {
 				root := c.Root()
-				projectId, err = helper.CreateTestProject("metal-cli-create-pro")
+				projectName := "metal-cli-create-pro" + helper.GenerateUUID()
+				projectId, err = helper.CreateTestProject(t, projectName)
+				t.Cleanup(func() {
+					if err := helper.CleanTestProject(t, projectId); err != nil &&
+						!strings.Contains(err.Error(), "Not Found") {
+						t.Error(err)
+					}
+				})
 				if err != nil {
 					t.Error(err)
+					return
 				}
+
 				if len(projectId) != 0 {
 
-					root.SetArgs([]string{subCommand, "create", "-p", projectId, "-P", "m3.small.x86", "-m", "da", "-O", "ubuntu_20_04", "-H", "metal-cli-create-dev"})
+					deviceName := "metal-cli-create-dev" + helper.GenerateUUID()
```

Comment:
The random string should be generated once at the start of a test, and the same value should be used in all resource names for the test (so the test project and the test devices in that project should have the same random string in their names).

Reply:
Right. Fixed